src/transports/dash/manifest_parser.ts

```diff
@@ -177,6 +177,17 @@ export default function generateManifestParser(
           });
           return processMpdParserResponse(value.continue(resources));
         }
+      }).catch((err) => {
+        const msgError = err instanceof Error ? err.message : "An unknown error has occured."
+        if (value.format === "string") {
+          return processMpdParserResponse(value.continue([
+            { responseData: "", error: { success: false, message: msgError } }
+          ]));
+        }
+        return processMpdParserResponse(value.continue([
+          { responseData: new ArrayBuffer(0),
```

Comment:
Isn't it strange to create a false response just because the type needs it?
Are you porting Go concept to JS :p?